api/all/src/main/java/io/opentelemetry/api/trace/propagation/W3CTraceContextPropagator.java

```diff
@@ -133,6 +133,14 @@ public static W3CTraceContextPropagator getInstance() {
       // No need to add an empty "tracestate" header.
       return;
     }
+    setter.set(carrier, TRACE_STATE, encodeTraceState(traceState));
+  }
+
+  /** Return the trace state encoded as a string according to the W3C specification. */
+  public static String encodeTraceState(TraceState traceState) {
```

Comment:
Let's move this to a new class, `io.opentelemetry.api.trace.propagation.internal.W3CTraceContextEncoding`. We don't want to expose this as a public API for users, at least not yet.

Reply:
Makes sense.